Source/common/SNTFileInfo.m

```diff
@@ -712,17 +713,15 @@ - (NSString *)resolvePath:(NSString *)path bundle:(NSBundle **)bundle {
 
 ///
 ///  Cache and return a MOLCodeSignChecker for the given file.  If there was an error creating the
-///  code sign checker, the error may be obtained from the read-only codesignCheckerError property.
+///  code sign checker it will be returned in the passed-in error parameter.
 ///
-- (MOLCodesignChecker *)codesignChecker {
-  @synchronized (self) {
-    if (!self.cachedCodesignChecker) {
-      NSError *error;
-      self.cachedCodesignChecker = [[MOLCodesignChecker alloc]
-          initWithBinaryPath:self.path error:&error];
-      _codesignCheckerError = error;
-    }
+- (MOLCodesignChecker *)codesignCheckerWithError:(NSError **)error {
+  if (!self.cachedCodesignChecker) {
```

Comment:
If the file is not codesigned do we want to enter this block if called multiple time?
Consider cashing an [NSNull null] object in self.cachedCodesignChecker if the file is not codesigned.
Then when returning, check the class type and return nil for NSNull objects.